perllib/FixMyStreet/App/Controller/Report/New.pm

```diff
@@ -1072,11 +1072,13 @@ sub process_report : Private {
         'may_show_name',                         #
         'subcategory',                              #
         'partial',                               #
-        'service',                               #
         'non_public',
       );
     $params{category} = $c->stash->{category};
 
+
+    my $service = ($c->sessionid ? $c->session->{app_platform} : $c->get_param('service')) || '';
+
```

Comment:
This would need to be `|| 'desktop'` to pass the cypress test - and presumably it's correct that this is where we need to assert desktop if it's not an app?

Reply:
Ooh, yes, good spot. I think I've got the logic wrong. What I was trying to say was "Use `app_platform` from session if it's there, otherwise the `service` param or `''`", but what I've actually written is "If there's a session then use `app_platform` or `''`, if no session then use the `service` param or `''`".

There was no mention of `desktop` before so I don't think that's necessary. I will rejig the logic to be correct and put a explanatory comment in.